src/type.js

```diff
@@ -1,650 +1,338 @@
-if(Sk.builtin === undefined) {
-    Sk.builtin = {};
-}
-
 /**
- * Maps Python dunder names to the Skulpt Javascript function names that
- * implement them.
- *
- * Note: __add__, __mul__, and __rmul__ can be used for either numeric or
- * sequence types. Here, they default to the numeric versions (i.e. nb$add,
- * nb$multiply, and nb$reflected_multiply). This works because Sk.abstr.binary_op_
- * checks for the numeric shortcuts and not the sequence shortcuts when computing
- * a binary operation.
+ * @namespace Sk.builtin
  *
- * Because many of these functions are used in contexts in which Skulpt does not
- * [yet] handle suspensions, the assumption is that they must not suspend. However,
- * some of these built-in functions are acquiring 'canSuspend' arguments to signal
- * where this is not the case. These need to be spliced out of the argument list before
- * it is passed to python. Array values in this map contain [dunderName, argumentIdx],
- * where argumentIdx specifies the index of the 'canSuspend' boolean argument.
- *
- * @type {Object}
+ * @description
+ * All the builtin types as well as useful functions
  */
-Sk.dunderToSkulpt = {
-    "__eq__": "ob$eq",
-    "__ne__": "ob$ne",
-    "__lt__": "ob$lt",
-    "__le__": "ob$le",
-    "__gt__": "ob$gt",
-    "__ge__": "ob$ge",
-    "__hash__": "tp$hash",
-    "__abs__": "nb$abs",
-    "__neg__": "nb$negative",
-    "__pos__": "nb$positive",
-    "__int__": "nb$int_",
-    "__long__": "nb$lng",
-    "__float__": "nb$float_",
-    "__add__": "nb$add",
-    "__radd__": "nb$reflected_add",
-    "__sub__": "nb$subtract",
-    "__rsub__": "nb$reflected_subtract",
-    "__mul__": "nb$multiply",
-    "__rmul__": "nb$reflected_multiply",
-    "__div__": "nb$divide",
-    "__rdiv__": "nb$reflected_divide",
-    "__floordiv__": "nb$floor_divide",
-    "__rfloordiv__": "nb$reflected_floor_divide",
-    "__invert__": "nb$invert",
-    "__mod__": "nb$remainder",
-    "__rmod__": "nb$reflected_remainder",
-    "__divmod__": "nb$divmod",
-    "__rdivmod__": "nb$reflected_divmod",
-    "__pow__": "nb$power",
-    "__rpow__": "nb$reflected_power",
-    "__contains__": "sq$contains",
-    "__bool__": ["nb$bool", 1],
-    "__nonzero__": ["nb$nonzero", 1],
-    "__len__": ["sq$length", 1],
-    "__get__": ["tp$descr_get", 3],
-    "__set__": ["tp$descr_set", 3]
-};
-
-Sk.setupDunderMethods = function (py3) {
-    if (py3) {
-        Sk.dunderToSkulpt["__matmul__"] = "tp$matmul";
-        Sk.dunderToSkulpt["__rmatmul__"] = "tp$reflected_matmul";
-    } else {
-        if (Sk.dunderToSkulpt["__matmul__"]) {
-            delete Sk.dunderToSkulpt["__matmul__"];
-        }
-        if (Sk.dunderToSkulpt["__rmatmul__"]) {
-            delete Sk.dunderToSkulpt["__rmatmul__"];
-        }
-    }
-};
+if (Sk.builtin === undefined) {
```

Comment:
it might be good to move this somewhere else, it's a bit odd it lives here.

Reply:
what about adding this to `/utils.js` - i presume it's defined here since this is the first place that the `Sk.builtin` namespace is used...